Tests/StreamChatTests/Workers/ChatClientUpdater_Tests.swift

```diff
@@ -215,113 +230,181 @@ final class ChatClientUpdater_Tests: XCTestCase {
+        let currentUserId: UserId = .unique
+        let initialToken: Token = .unique(userId: currentUserId)
+        let updatedToken: Token = .unique(userId: currentUserId)
 
-    func test_reloadUserIfNeeded_happyPaths() throws {
-        struct Options {
-            let initialToken: Token
-            let updatedToken: Token
+        // Create an active client with user session.
+        let client = mockClientWithUserSession(token: initialToken)
+        
+        // Create and track channel controller
+        let channelController = ChatChannelController(
+            channelQuery: .init(cid: .unique),
+            channelListQuery: nil,
+            client: client
+        )
+        client.trackChannelController(channelController)
+        
+        // Create and track channel list controller
+        let channelListController = ChatChannelListController(
+            query: .init(filter: .exists(.cid)),
+            client: client
+        )
+        client.trackChannelListController(channelListController)
+        
+        // Create an updater.
+        let updater = ChatClientUpdater(client: client)
+        
+        // Save current background worker ids.
+        let oldWorkerIDs = client.testBackgroundWorkerId
+        
+        // Simulate `reloadUserIfNeeded` call.
+        var reloadUserIfNeededCompletionCalled = false
+        var reloadUserIfNeededCompletionError: Error?
+        updater.reloadUserIfNeeded(
+            userConnectionProvider: .static(updatedToken)
+        ) {
+            reloadUserIfNeededCompletionCalled = true
+            reloadUserIfNeededCompletionError = $0
         }
-
+        
+        // Assert current user id is valid.
+        XCTAssertEqual(client.currentUserId, updatedToken.userId)
+        // Assert token is valid.
+        XCTAssertEqual(client.currentToken, updatedToken)
+        // Assert web-socket is not disconnected
+        XCTAssertEqual(client.mockWebSocketClient.disconnect_calledCounter, 0)
+        // Assert web-socket endpoint is valid.
+        XCTAssertEqual(
+            client.webSocketClient?.connectEndpoint.map(AnyEndpoint.init),
+            AnyEndpoint(.webSocketConnect(userInfo: UserInfo(id: updatedToken.userId)))
+        )
+        // Assert `completeTokenWaiters` was called.
+        XCTAssertTrue(client.completeTokenWaiters_called)
+        // Assert `completeTokenWaiters` was called with updated token.
+        XCTAssertEqual(client.completeTokenWaiters_token, updatedToken)
+        // Assert background workers stay the same.
+        XCTAssertEqual(client.testBackgroundWorkerId, oldWorkerIDs)
+        // Assert database is not flushed.
+        XCTAssertFalse(client.mockDatabaseContainer.removeAllData_called)
+        // Assert active components are preserved
+        XCTAssertTrue(client.activeChannelControllers.contains(channelController))
+        XCTAssertTrue(client.activeChannelListControllers.contains(channelListController))
```

Comment:
Both the tests and the logic can be simplified a lot if all the actions performed on properties of the ChatClient are under one method. Less knowledge, more testable. And this particular logic to be performed when this event happens should be tested from the ChatClient, then